src/core/connection.c

```diff
@@ -5570,6 +5573,8 @@ QuicConnRecvDatagrams(
             DatagramPath->Route.Queue != Datagram->Route->Queue) {
             DatagramPath->Route.Queue = Datagram->Route->Queue;
         }
+
+        CxPlatUpdateRouteTcpState(&DatagramPath->Route, Datagram->Route);
```

Comment:
Can we somehow refactor this, plus the code above (maybe), to be generic, non-TCP specific code?